apps/web/src/fetcher/create-breadcrumbs.tsx

```diff
@@ -7,96 +7,69 @@ import {
-      name = current.title
-      current = current.parent
-    }
-    return name
+  const breadcrumbs = term.path
+    .filter((entry) => entry?.id !== 3) // TODO: maybe remove in API
+    .map((entry) => {
+      return {
+        label: entry!.title,
+        url: entry!.alias,
+        id: entry!.id,
+      }
+    })
+
+  if (includeFirstParent) {
+    breadcrumbs.push({
+      label: term.title,
+      url: term.alias,
+      id: term.id,
+    })
   }
 
-  while (current.id !== 3 && current.parent) {
-    // find subject of this path and rewrite root breadcrumb
-    const matching = secondaryMenus.filter((menu) => menu.rootId === current.id)
-    if (matching.length > 0) {
-      breadcrumbs.unshift({
-        label: matching[0].rootName ?? current.title,
-        url: matching[0].landingUrl,
-        id: current.id,
-      })
-      break // not going further, especially for subjects under construction
-    } else {
-      // check if this breadcrumb is already part of secondary menu
-      const matching2 = secondaryMenus.filter((menu) =>
-        menu.entries.some((entry) => entry.id === current.id)
-      )
-      if (matching2.length > 0) {
-        const metaMenuEntry = matching2[0].entries.filter(
-          (menu) => menu.id === current.id
-        )[0]
-
-        breadcrumbs.unshift({
-          label: metaMenuEntry.title,
-          url: current.alias,
-          id: current.id,
-        })
-        breadcrumbs.unshift({
-          label: matching2[0].rootName ?? getSubjectName(current),
-          url: matching2[0].landingUrl,
-          id: matching2[0].rootId,
-        })
-        break
-      } else {
-        breadcrumbs.unshift({
-          label: current.title,
-          url: current.alias,
-          id: current.id,
-        })
-      }
-    }
-    // the recursion is limited, so there is a slight type mismatch
-    current = current.parent
+  const subject = secondaryMenus.find(
+    (menu) => menu.rootId === breadcrumbs[0].id
+  )
+
+  if (subject) {
+    breadcrumbs[0].label = subject.rootName ?? breadcrumbs[0].label
+    breadcrumbs[0].url = subject.landingUrl ?? breadcrumbs[0].url
   }
 
   return breadcrumbs
+
+  // TODO: check if we still need this
+  //     // check if this breadcrumb is already part of secondary menu
+  //     const matching2 = secondaryMenus.filter((menu) =>
+  //       menu.entries.some((entry) => entry.id === current.id)
+  //     )
+  //     if (matching2.length > 0) {
+  //       const metaMenuEntry = matching2[0].entries.filter(
+  //         (menu) => menu.id === current.id
+  //       )[0]
+  //       breadcrumbs.unshift({
+  //         label: metaMenuEntry.title,
+  //         url: current.alias,
+  //         id: current.id,
+  //       })
+  //       breadcrumbs.unshift({
+  //         label: matching2[0].rootName ?? getSubjectName(current),
+  //         url: matching2[0].landingUrl,
+  //         id: matching2[0].rootId,
+  //       })
+  //       …
```

Comment:
@Entkenntnis tbh. I don't really understand why this is / was needed.
Do you remember?

Reply:
This affects situations like `FOS & BOS` where taxonomies are part of the secondary menu and need some kind of replacement.

![grafik](https://github.com/serlo/frontend/assets/13507950/6ba3090c-9063-4762-b37d-cbe146c2c79a)

Reply:
If you don't do it, you will get the complete path

root > Mathe > Deutschland  > Bayern > `FOS & BOS`

Reply:
Thanks for the explanation!